PyPoE/poe/file/psg.py

```diff
@@ -270,15 +270,23 @@ def _read(self, buffer, *args, **kwargs):
         version = struct.unpack_from('<B', data, offset=offset)[0]
         offset += 1
 
-        unknown_length = struct.unpack_from('<B', data, offset=offset)[0]
+        unknown_length = 8 
+        #We used to be able to fetch the count of throwaway unknown data from the start of the .psg, but it doesn't work
+        #as of 3.16. Manually, I looked for where there's a 32 bit unsigned int equal to 7, and ignored everything before it.
+        #unknown_length = struct.unpack_from('<B', data, offset=offset)[0]
         offset += 1
 
+
         unknown = struct.unpack_from(
             '<' + 'B'*unknown_length, data, offset=offset
         )
         offset += 1*unknown_length
 
         root_length = struct.unpack_from('<I', data, offset=offset)[0]
+        if(root_length > 1000):
+            raise ValueError(
+                'root_length is unrealistically large at {}.\nStopping to prevent allocating too much memory'.format(root_length)
```

Comment:
https://github.com/Project-Path-of-Exile-Wiki/PyPoE/pull/37#discussion_r744102703

Reply:
Fixing